app/controllers/scans_controller.rb

```diff
@@ -27,20 +27,11 @@ def new_request_params
   end
 
   def redirect_to_success_with_token
-    redirect_to illiad_query(current_request)
+    redirect_to illiad_url
   end
 
-  def illiad_query(scan)
-    illiad_params = {
-      'Action': '10', 'Form': '30', 'rft.genre': 'scananddeliverArticle',
-      'rft.jtitle': scan.item_title,
-      'rft.au': scan.data[:authors],
-      'rft.pages': scan.data[:page_range],
-      'rft.atitle': scan.data[:section_title],
-      'rft.volume': scan.holdings.first.callnumber,
-      'scan_referrer': successful_scan_url(scan)
-    }
-    Settings.sul_illiad + "#{illiad_params.to_query}"
+  def illiad_url
+    @illiad_url ||= IlliadOpenurl.new(current_request, successful_scan_url(current_request)).to_url
```

Comment:
One way to make this method more testable would be to pass the `current_request` in as a parameter, like we were doing in `#illiad_query`.

It doesn't look like we call `#illiad_url` more than once in a request, so the memoization may be unnecessary too.